openglyph/src/utility/steam_paths.cpp

```diff
@@ -0,0 +1,119 @@
+#include "vdf_fileparser.hpp"
+
+#include <khepri/io/exceptions.hpp>
+#include <khepri/utility/platform.hpp>
+
+#include <openglyph/utility/steam_paths.hpp>
+
+#include <iostream>
+namespace openglyph::utility {
+
+/// @brief Returns the usual Steam root installation path for the current platform.
+/// @return The root Steam installation directory as a std::filesystem::path.
+std::filesystem::path get_steam_root_path()
+{
+    switch (khepri::utility::get_current_platform()) {
+    case khepri::utility::Platform::windows:
+#if defined(_WIN32) || defined(_WIN64)
+    {
+        std::filesystem::path steampath =
+            khepri::utility::get_registry_key(HKEY_LOCAL_MACHINE,
+                                              "SOFTWARE\\WOW6432Node\\Valve\\Steam", "InstallPath")
+                .value_or("C:\\Program Files (x86)\\steam");
+        if (!std::filesystem::exists(steampath)) {
+            throw khepri::io::FileNotFoundError();
+        }
+
+        if (!std::filesystem::exists(steampath / "steam.exe")) {
+            throw khepri::io::FileNotFoundError();
+        }
+        return steampath;
+    }
+#else
+        return "C:\\Program Files (x86)\\steam";
+#endif
+    case khepri::utility::Platform::linux: {
+        std::filesystem::path steampath = "~/.steam";
+
+        if (!std::filesystem::exists(steampath)) {
```

Comment:
Redundant check, as I indicated previously. If the file exists, the parent directory also exists.
Same issue for Windows.